velox/docs/develop/types.rst

```diff
@@ -73,6 +73,31 @@ Examples:
 * Timestamp(-5000*24*60*60 - 1000, 123456) represents 1956-04-24 07:43:20.000123456
   (5000 days 1000 seconds before epoch plus 123456 nanoseconds).
 
+Floating point types (REAL, DOUBLE) have special values negative infinity, positive infinity, and
+not-a-number (NaN).
+
+For NaN the semantics are different than the C++ standard floating point semantics:
+
+* NaN = NaN returns true.
```

Comment:
i would clarify also the behavior for NaN > NaN or NaN < NaN.  I assume that in cpp NaN < x and NaN > x return false.  And you probably want to have it follow the sort order.

Reply:
It sort of is described two sentences down. In CPP semantics NaN < x and NaN > x do return false. But in Velox semantics NaN > x would return true  as NaN is considered larger than any other value.
NaN < x is false in Velox too.

Reply:
Thanks for updating with that information. Wanted to explain that the reason i suggested stating it explicitly is that by IEEE semantics there is a total order defined in which +NaN sorts largest, but comparing NaN < x and NaN > x will always return false.